src/dotnet/Core/Text/StringExt.cs

```diff
@@ -19,6 +19,8 @@ public static string RequireNonEmpty(this string? source, string name)
     [return: NotNullIfNotNull("source")]
     public static string? RequireEmpty(this string? source, string name)
         => source.IsNullOrEmpty() ? source : throw StandardError.Constraint($"{name} must be null or empty here.");
+    public static string? RequireMaxLength(this string source, int length, string name)
+        => source.Length <= length ? source : throw StandardError.Constraint($"{name} must not be longer than {length}.");
```

Comment:
Maybe a better option: "Must be no more than X characters." (checked Google, 38M vs 1.4M results)

Reply:
done